### PR TITLE
Add test for read access to additional Core GitOps APIs

### PR DESCRIPTION
**Pull request by jgwest**
Updates test for readonly access for appstudio users to gitopsdeploymentmanagedenvironments, gitopsdeploymentrepositorycredentials, gitopsdeploymentsyncrun APIs.

Corresponding host-operator PR: https://github.com/codeready-toolchain/host-operator/pull/781
Corresponding book PR: https://github.com/redhat-appstudio/book/pull/90

### CODE REVIEW

**Review by ranakan19 — APPROVED** [review]

File: testsupport/tiers/checks.go [hunk 1]
```diff
@@ -1282,7 +1282,7 @@ func appstudioUserActionsRole() spaceRoleObjectsCheck {
 				},
 				{
 					APIGroups: []string{"managed-gitops.redhat.com"},
-					Resources: []string{"gitopsdeployments"},
+					Resources: []string{"gitopsdeployments", "gitopsdeploymentmanagedenvironments", "gitopsdeploymentrepositorycredentials", "gitopsdeploymentsyncruns"},
 					Verbs:     []string{"get", "list", "watch"},
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:1286]: you changed the template so the admin has all permissions
```suggestion
					Verbs:     []string{"*"},
```

Comment (alexeykazakov) [reply testsupport/tiers/checks.go:1286]: Good catch @MatousJobanek

**Comment by alexeykazakov** [comment]
A known issue with a flaky test. It's being investigated. See https://redhat-internal.slack.com/archives/CHK0J6HT6/p1680900910350839 for more details.

/retest

**Pull request merged**